### ISSUE TITLE
HUU: Design: Add note-taking pop up in Coordinator Dashboard

### ISSUE THREAD

**Issue by ExperimentsInHonesty** [hackforla/HomeUniteUs#736]
### Overview
We need the coordinators to be able to write/save notes about the applicant or the progress.  So that they can use the notes for any kind of information that is not already covered by the MVP design of the system

### Action Items
- [ ]  Review Existing Design
- [ ]  Updating Design as Needed
- [ ]  Designs are edited and ready for engineering. Link to the Figma: (share Figma link here when ready)

### Resources/Instructions
- Product Requirements/User Story: https://github.com/hackforla/HomeUniteUs/issues/519
- [Figma design](https://www.figma.com/file/BNWqZk8SHKbtN1nw8BB7VM/HUU-Everything-Figma-Nov-2022?type=design&node-id=12018-4783&mode=design&t=GSKQXvW2wV1WXfeu-0#756597241)

**Comment by ExperimentsInHonesty** [comment]
If they only have one notes field, and can add or remove something from it... then having a number inside the icon, dosent make sense.  I recommend the following
If there is a note, the icon is solid with a check mark in it
if there are no notes, its open and has the + 1 on it.

![image](https://github.com/user-attachments/assets/48709014-9a70-480f-bc8b-4fdea74a4612)


I also noticed that the notes field was grayed out on the table view... is there a reason why they could not click on the note +1 icon from that view and leave a note then?

**Comment by lasryariel** [comment]
added comment to one of the older existing designs in figma, we should revert to those versions for MVP for simplicity. 

https://www.figma.com/design/BNWqZk8SHKbtN1nw8BB7VM/HUU-Everything-Figma-Nov-2022?node-id=15207-18315&t=PtTctlrVruzua57D-4

**Comment by lasryariel** [comment]
Updated parent comment with design team questions pulled from Figma including answers from product and other feedback.

Here is a screenshot of one of the frames in the current wireframe:
![image](https://github.com/user-attachments/assets/6387daea-7852-40f5-8040-f1495598d5a3)

**Comment by lasryariel** [comment]
Notes from 9/10 breakout session:

- add profile pic icon to sticky section
- what does expand look like while still in side view
- separate intake profile approval from overall approval
- allow multiple tabs of accordion to be opened - mostly a stakeholder question
- define action items
- width of expansion screens - need to think about it, depends on width of details. maybe steps should be collapsible
- keep scalability in mind- design should easily adapt to more onboarding events, more intake profile section

**Comment by lasryariel** [comment]
Ariel's notes from Coordinator breakout session 9/17/24 - to be combined with Gabriella's notes

- Decision made to set aside the sidebar designs for now. Complete the full screen designs/wireframes for each section, and then we will see how a sidebar can be useful.
  - This decision was made because including the full information for each step in the sidebar view is too much, but we could not clearly articulate what abbreviated information will actually be useful. This can potentially be a post-mvp feature.
- Design team will focus on full screen and work on hierarchy of sections
  - Steps should be clearly sequential
  - Notes should be in an obvious and accessible section

Intake Profile approval:
- need screen of how coordinator adds approval notes
  - when looking at prior versions, the approval notes can potentially be in the same spot that the coordinator adds approval notes

Intake Profile Version history:
- Approval Notes for each version should be next to each other in any version history view
- the current version should show in the list for version history
- version should show approval date as the "at a glance" date

Onboarding Events:
- Bonnie has provided Jasika updated information on how the onboarding events will work. Need to sync with her and update #635 accordingly.

**Comment by gabcdominic** [comment]
**10/1/2024 Updates to UI/UX and Features:**

**1. Hierarchy Removal & Color-Coding**
- Removed the top hierarchy structure.
- Introduced color-coding for different states (e.g., “Accepted” status highlighted in green) for quick recognition.

**2. Version History Display**
- Removed the icon and placed the version history action next to the date for improved alignment and visual clarity. 
![Screen Shot 2024-10-01 at 7 47 09 PM](https://github.com/user-attachments/assets/1a1a7994-02e5-4fe7-b693-029b5cb016e6)

![Screen Shot 2024-10-01 at 8 21 40 PM](https://github.com/user-attachments/assets/e4b35805-5f93-4002-986f-90ea936ae27c)

**3. Notes**
- We prefer the sidebar layout over other options.
- Notes will resemble a chat-style pop-up for a modern and interactive feel, making them easy to engage. The goal is to ensure easy access while maintaining a clean, uncluttered interface.

**Comment by gabcdominic** [comment]
**10/15/24 Meeting Notes** 

Design updated -- [new version](https://www.figma.com/design/BNWqZk8SHKbtN1nw8BB7VM/HUU-Everything-Figma-Aug-2024?node-id=2353-17341&node-type=canvas&t=0azZhLCeIAzc5RYH-0) reflects changes below. 

**Pop up notification**
- When additional information is requested or available, a pop-up notification will appear on the screen, allowing coordinators to see the update immediately. 
![Screen Shot 2024-10-15 at 8 03 59 PM](https://github.com/user-attachments/assets/4a74d24c-dc22-49a4-bc2c-e3f6560a1eec)

**Notes**
- _User Perception_: Upon further thought, notes should not mimic a chat box as it could lead to misinterpretation. Notes should be distinct from a chat box to avoid confusion with Host/Guest chat or AI features. The visual must indicate it’s for profile notes. Future discussions may explore a 'post-it note' aesthetic for clarity. The notes icon is placed next to the individual's name to prevent any confusion with a chatbot interface.
![Insert](https://github.com/user-attachments/assets/2d3fffc9-67ab-45ab-916b-27b3cf6f9635)


**Mobile Design**
- Nihan is developing a mobile version that mirrors the desktop design. Icons and language will remain consistent across platforms
![Screen Shot 2024-10-15 at 7 20 31 PM](https://github.com/user-attachments/assets/a2b9b044-d2aa-4806-bf7e-e76915ae4e51)


**Next Steps**
- Present the design to the internal team by **10/22/24**
- Stakeholder review with Bonnie on **10/29/24**

**Comment by gabcdominic** [comment]
**10/22/2024**

**Link to Figma** [here](https://www.figma.com/proto/BNWqZk8SHKbtN1nw8BB7VM/HUU-Everything-Figma-Aug-2024?page-id=2353%3A17341&node-id=16449-6418&node-type=frame&viewport=1320%2C2110%2C0.85&t=C5wVOPysXpfaAh4x-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=16449%3A6418).

- Version History: Essential for tracking feedback and measuring improvements. Ensure clear version history to compare feedback, especially if multiple revisions to a profile exist. We will think about this post MVP (ex. what if there are ten revisions? Tracking progress remains key to managing stakeholder expectations). 
- Profile Navigation: Implement up/down arrows for profile navigation. Ensure it aligns with the overall product flow. 
- Notes Overlay Panel: Replace the "chat interface" look with an overlay panel to avoid confusion. It's notes, not chat.
- Profile-Level Notes: Notes are tied to the profile level
- Notification MVP: For MVP, implement a _one-time_ pop-up notification (shown below). Keep it simple.
- Wireframe Focus: We need the layout to make sense now. Ensure wireframes prioritize logical layout and clarity. Once this is done, move to optimize the information flow.
- Development Frame Addition: Add a frame on the development website with a click-through feature for Sophia Johnson. It should communicate the feature’s purpose to users. Priyanka will review the wireframes on Sunday. 

![Screen Shot 2024-10-24 at 5 11 18 PM](https://github.com/user-attachments/assets/3e53ea23-4342-4ea7-bbe9-92562e5f4ce9)

**Comment by lola3736** [comment]
@lasryariel okay to remove the label "ready for product" as it looks like Design is currently working on the wireframe with discussions with Product.

**Comment by gabcdominic** [comment]
### 10/29/24 Stakeholder Meeting

Layout approved. Next, we need engineering validation on feature viability to determine prioritization and avoid churn, scope creep, and design rework.

**MVP Decisions:**

1. **Navigation Flow**:  _Previous & Next Options_: Removed for MVP. After completing a section, Coordinators are directed back to the dashboard to streamline task navigation.
2. **Approval Process and Visual Changes**: Coordinators approve sections individually. Consider visible change highlighted in a _side-by-side view_ for easy comparison of "before" and "after" states (see below).
3. **Intake Profile Feedback:** No character limit for Coordinators on intake profile feedback, allowing flexibility without imposing restrictions.
4. **Version History and Display:**
- _Intake Profile View_: Stakeholder suggested for Coordinators to be able to see both current and prior versions side-by-side at both the top and bottom of the screen, facilitating seamless comparison.
Update Requests: 
- _End-User View_ (Hosts/Guests): Only the latest update request from the Coordinator is visible, ensuring simplicity in their view.
- _Coordinator View_: Displays a complete chronological history, with the latest entries at the top. This enables Coordinators to manage updates by keeping relevant notes and removing those no longer needed.
5. **Completion and Updates Process**: _Notes and Requested Changes_: After each section, Coordinators can add notes to request changes. Once action is taken on the requested change, the note disappears from the Host/Guest view but remains in the historical log for the Coordinator. Hosts/Guests see "changes requested—see details," showing the original question and the Coordinator’s feedback, which can be displayed as a thought bubble or feedback box.
6. **Enhanced Notes Features**:
- Coordinators can view summaries of notes with the option to collapse sections for a streamlined experience. By default, notes are fully visible, but users can opt to hide them as needed.
- Edit/Delete Capability: Coordinators can edit or delete notes to ensure accurate and relevant information remains visible.
7. **Search Capability:** Coordinators will have a search function for profiles and notes, making it easier to find specific information efficiently.

**Post-MVP Considerations:**
1. **Previous & Next Options**: What happens to Previous & Next options if a filter is applied on the dashboard? 
2. **Admin Oversight for Notes:** To prevent potential issues (e.g., inappropriate content), admin oversight can be added post-MVP, allowing admins to edit/remove specific notes if necessary.

![Slide1](https://github.com/user-attachments/assets/57c4542b-1aba-45a0-917f-5c1c282b8454)

**Comment by gabcdominic** [comment]
### 11/12/2024 Team Meeting

1. **Guest/Host Team Sync Needed** to define the intake profile submission and resubmission process. For example, UI expectations for handling unapproved fields, enabling feedback loops and revisions for intake profiles, and locking approved fields in the Guest/Host UI to prevent modification. For example, if there's revision needed under "Substance Abuse," can that section of the intake profile be editable, and we lock all other sections that have been approved?
![Screen Shot 2024-11-12 at 7 08 15 PM](https://github.com/user-attachments/assets/5f966c31-eb41-40dc-afee-e39d33c31500)

2.** MVP Scope Reminder** to minimize Coordinator workload. Prioritize elements that streamline their tasks.

3. **Discussions regarding 10/29/24 stakeholder feedback**
- _Navigation (Next/Previous Buttons)_: Feasible implementation for us to have next/previous functionality to fetch data in sets. Engineering will have to look more into this if we have large data sets (ex. 1-50 are on page 1, 51, and on is on page 2 of the dashboard)
- _Side-by-side before and after layout_ should be workable. Line-by-line design can be too complex (feasibility and ease of development)
- _Notes management (Search/Edit/Delete)_: confirmed to be implementable, can proceed
- _Feature prioritization_: 1) Intake Profile 2) Notes Management. Execution: Priyanka to lead Intake Profile, Nihan to work on Notes. Engineering begins overall layout
- _Dashboard customization (saved view dashboard customization, shared views between coordinators for filters applied)_: We need to assess further technical requirements for saved views and coordinator-specific shared views based on applied filters
![Screen Shot 2024-11-12 at 7 34 24 PM](https://github.com/user-attachments/assets/35fabec7-13d5-4228-b10a-97d6b4ff4a50)

4. **Engineering Parallel Development Approach**
- The objective is to allow Engineering to progress with skeleton and iterative refinements. Engineering can start with general structural design with minimal expected changes like the approved core layout, user profile views, and accordions for content organizations. They can maintain the current scaffolding & adjust only if necessary. 
- Engineering will need more direction for the intake profile as we work with Host/Guest on intake profile submission & resubmissions

**Comment by gabcdominic** [comment]
Design updated work on side-by-side Version History (Work in Progress)

![Image](https://github.com/user-attachments/assets/3ea79363-b3c9-4646-a643-09d86c0c9a55)

![Image](https://github.com/user-attachments/assets/ccfb11fb-814d-4824-b91a-1d1bfb983166)

**Comment by gabcdominic** [comment]
- Progress was made on design; there is potential for having **track changes**. Feasibility for MVP presents execution challenges contingent on engineering feasibility and prioritization.
- Resource Update: Priyanka from Design is stepping back from volunteering with Hack for LA. Nihan is still on Coordinator Design. Jaime is working on Design System efforts for HUU. FYI @lasryariel 

![Image](https://github.com/user-attachments/assets/14b6f542-2380-4a20-986c-b4dc94bb23dc)

**Comment by gabcdominic** [comment]
![Image](https://github.com/user-attachments/assets/45c66ade-0864-484b-9fcb-a90ac9706a59)

**1. One-Time vs. Ongoing Feature:** A single request upfront ensures all necessary details are collected, but it may create issues if missing or unclear information arises later.
**2. Until Application is Accepted:** Multiple requests should be allowed during verification and matching to ensure Coordinators can clarify details without blocking progress. Requests can be locked once the application is accepted or capped at a reasonable limit (e.g., two per phase).
**3. Always Available Option:** Keeping this open indefinitely may lead to excessive back-and-forth, disrupting the user experience.

My recommendation: Option 2

- Allow requests throughout intake but lock them after acceptance for efficiency.
- Set a threshold (e.g., two rounds of requests) to minimize delays.

Thoughts @lasryariel @sanjumv

**Comment by gabcdominic** [comment]
### Tuesday, 3/4/25 -- Design Update for #731 

The onboarding event tracking system meets basic functional requirements (MVP 1 acceptance criteria) but still lacks key automation and tracking features (MVP 2 and 3?). 

**What We're Currently Solving**

1. Coordinator can easily find Onboarding Information for a Guest/Host.
2. Basic Event Management: Coordinators can update and edit event status, scheduled date, and completion date. 
3. The completion date remains editable.

![Image](https://github.com/user-attachments/assets/ed23c48c-ca7e-452d-bd9c-6a468801846f)

![Image](https://github.com/user-attachments/assets/b7873582-93b8-46a2-8e09-c5685db027dd)

**To consider**
Without Home Unite Us, coordinators manually handle all scheduling, which slows down onboarding and creates unnecessary back-and-forth communication. While the current system tracks dates at an organizational level, coordinators must still ensure each event is scheduled and completed—a time-consuming administrative task. The biggest gap is the lack of self-scheduling for Guests and Hosts. Without this, coordinators remain the bottleneck in the process.

_1. Need for a Scheduling Solution:_ We have not yet decided whether to use a third-party tool (Google Calendar Appointment Scheduler, Calendly, Jotform, SignUpGenius) or build an in-house solution. Without automation, coordinators must manually adjust schedules, and there’s no real-time sync between them, Guests, and Hosts. This leads to miscommunication and delays, making the system only slightly better than using a spreadsheet. A decision on this will impact design and engineering. 

_2. Lack of Automated Notifications:_ Currently, the system does not send confirmation emails or reminders when events are scheduled or completed. Coordinators must follow up manually, which is inefficient. MVP must include automated updates to ensure Guests and Hosts receive timely notifications without extra work from coordinators.

**Post-MVP Consideration**
After solving scheduling and notifications, we can explore tracking overdue events. Right now, if an event isn’t completed, the system doesn’t flag it, potentially causing delays. While important, this isn’t a blocker for MVP.

**Comment by lola3736** [comment]
@ExperimentsInHonesty we are awaiting your feedback on Coordinator notes, so we can queue this up for Dev, thanks much.

**Comment by lola3736** [comment]
@gabcdominic @nihandurmaz is this dependency label applicable?  I dont see any dependency item attached and it doesnt seem that there is a blocker here.
